blog/12-01-2023-connect-and-interact-with-celo-using-web3react/connect-and-interact-with-celo-using-web3react.md

```diff
@@ -0,0 +1,540 @@
+  }, [response, account]);
+```
+
+- `handleClick()` accepts an id of type number to programmatically select the choice of connector.
+
+```bash
+  const handleClick = async(index:number) => { 
+    setConnecting(true);
+    await handleConnect(index).then(() => {
+      setConnecting(false);
+    });
+  }
+```
+
+- We filter the contract methods for the ones we need using the `displayContractFunctions` function.
+
+```bash
+  const displayContractFunctions = useMemo(() => {
+    let filt: any;
+    if (!vaultAbi) return [];
+    filt = vaultAbi.filter(method => method["type"] === "function");
+    return filt.filter(
+      (method: { name: string }) => method.name === "stake" ||
+        method.name === "unstake" || 
+          method.name === "stakeOnBehalf" ||
+            method.name === "compoundStaking" || 
+              method.name === "getStakeProfile" ||
+                method.name === "withdraw"
+        );
+  }, [vaultAbi]);
+```
+
+- Inside the `handleSubmit` function, add a new case to the switch statement to handle the new feature.
+
+```bash
+case 'compoundStaking':
+  result = await stake();
+  break;
+```
+
+**components/api/index.tx**
+
+- Lastly, the functionalies in this file remain unchanged except that we added a new feature in the switch statement for the function we introduced in our contract.
+
+```bash
+  // >...
+  case 'compoundStaking':
+    const txn1 = await vault_ins.compoundStaking({value: value });
+    // const txn = await vault_ins.setToken(tokenAddr);
+    await txn1?.wait(2).then((rec: ContractReceipt) => {
+      result.receipt = rec;
+      result.view = false;
+      if(cancelLoading) cancelLoading();
+    });
+    break;
+  // >...
+```
+
+In case you missed any or confused with the explanation, the complete source code can be found **[here](https://github.com/bobeu/connect-and-interact-with-celo-using-web3react)**.
+
+
+## Conclusion​
+
+What we have learned so far:
+
+- Writing, compiling, deploying and testing smart contracts using foundry.
+- How to install, set up, and use web3React library.
+- How to build a simple frontend and connect the backend.
+- Manage your dApp using web3React.
+
+## What next?
+​
+You can edit the code to your taste. At this point, you should be able to launch your own project as Celo developer. If you are confused anywhere in the tutorial, I left a link to the full source code at the _references_ section. Go over it as many times as you can. In no time, you will get a good grasp. Celo is EVM-compatible blockchain. You can leverage the various web3 tools to build your dream application. Get started with **[Celo documentation](https://docs.celo.org/tutorials)**
+
+## About the Author​
+
+**Isaac Jesse** , aka _Bobelr_ is a smart contract/Web3 developer. He has been in the field since 2018, worked as an ambassador with several projects like Algorand and so on as content producer. He has also contributed to Web3 projects as a developer.
```

Comment:
not necessary but you might want to consider adding a link to you linkedIn profile